pkg/devicerepository/store/remote/schema.go

```diff
@@ -0,0 +1,460 @@
+	SupportsJoin              bool                `yaml:"supportsJoin"`
+	Rx1Delay                  ttnpb.RxDelay       `yaml:"rx1Delay"`
+	Rx1DataRateOffset         uint32              `yaml:"rx1DataRateOffset"`
+	Rx2DataRateIndex          ttnpb.DataRateIndex `yaml:"rx2DataRateIndex"`
+	Rx2Frequency              float64             `yaml:"rx2Frequency"`
+	FactoryPresetFrequencies  []float64           `yaml:"factoryPresetFrequencies"`
+	MaxEIRP                   float32             `yaml:"maxEIRP"`
+	MaxDutyCycle              float64             `yaml:"maxDutyCycle"`
+	Supports32BitFCnt         bool                `yaml:"supports32bitFCnt"`
+}
+
+var errRegionalParametersVersion = errors.DefineNotFound("regional_parameters_version", "unknown Regional Parameters version `{phy_version}`")
+
+// ToTemplatePB returns a ttnpb.EndDeviceTemplate from an end device profile.
+func (p EndDeviceProfile) ToTemplatePB(ids *ttnpb.EndDeviceVersionIdentifiers, info *ttnpb.EndDeviceModel_FirmwareVersion_Profile) (*ttnpb.EndDeviceTemplate, error) {
+	phyVersion, ok := regionalParametersToPB[p.RegionalParametersVersion]
+	if !ok {
+		return nil, errRegionalParametersVersion.WithAttributes("phy_version", p.RegionalParametersVersion)
+	}
+
+	paths := []string{
+		"version_ids",
+		"supports_join",
+		"supports_class_b",
+		"supports_class_c",
+		"lorawan_version",
+		"lorawan_phy_version",
+	}
+	dev := ttnpb.EndDevice{
+		VersionIDs:        ids,
+		SupportsJoin:      p.SupportsJoin,
+		SupportsClassB:    p.SupportsClassB,
+		SupportsClassC:    p.SupportsClassC,
+		LoRaWANVersion:    p.MACVersion,
+		LoRaWANPHYVersion: phyVersion,
+	}
+
+	if info.CodecID != "" {
+		dev.Formatters = &ttnpb.MessagePayloadFormatters{
+			DownFormatter: ttnpb.PayloadFormatter_FORMATTER_REPOSITORY,
+			UpFormatter:   ttnpb.PayloadFormatter_FORMATTER_REPOSITORY,
+		}
+		paths = append(paths, "formatters")
+	}
+
+	dev.MACSettings = &ttnpb.MACSettings{}
+	if p.ClassBTimeout > 0 {
+		t := time.Duration(p.ClassBTimeout) * time.Second
+		dev.MACSettings.ClassBTimeout = &t
+		paths = append(paths, "mac_settings.class_b_timeout")
+	}
+	if p.ClassCTimeout > 0 {
+		t := time.Duration(p.ClassCTimeout) * time.Second
+		dev.MACSettings.ClassCTimeout = &t
+		paths = append(paths, "mac_settings.class_c_timeout")
+	}
+	if p.PingSlotDataRateIndex > 0 {
+		dev.MACSettings.PingSlotDataRateIndex = &ttnpb.DataRateIndexValue{
+			Value: p.PingSlotDataRateIndex,
+		}
+		paths = append(paths, "mac_settings.ping_slot_data_rate_index")
+	}
+	if p.PingSlotFrequency > 0 {
+		dev.MACSettings.PingSlotFrequency = &pbtypes.UInt64Value{
+			Value: uint64(p.PingSlotFrequency * 100000),
+		}
+		paths = append(paths, "mac_settings.ping_slot_frequency")
+	}
+	if p.PingSlotPeriod > 0 {
+		dev.MACSettings.PingSlotPeriodicity = &ttnpb.PingSlotPeriodValue{
+			Value: pingSlotPeriodToPB[p.PingSlotPeriod],
+		}
+		paths = append(paths, "mac_settings.ping_slot_periodicity")
+	}
+	if p.Rx1Delay > 0 {
```

Comment:
`0` (mapping to `1`) is a valid value I suppose @johanstokking ?

Reply:
Yes, that is a valid value that defaults to 1